packages/pod-install/src/index.ts

```diff
@@ -19,21 +19,43 @@ const program = new Command(packageJSON.name)
   .option('--quiet', 'Only print errors')
   .option('--non-interactive', 'Disable interactive prompts')
   .allowUnknownOption()
-  .parse(process.argv);
+  .parse(process.argv)
+  .action(async (projectDirectory?: string) => {
+    try {
+      await runAsync(projectDirectory);
+      if (!program.opts().quiet) {
+        await shouldUpdate();
+      }
+    } catch (reason: any) {
+      console.log('\nAborting run');
+      if (reason.command) {
+        console.log(`  ${chalk.magenta(reason.command)} has failed.`);
+      } else {
+        console.log(chalk.red`An unexpected error was encountered. Report it on GitHub: https://github.com/expo/expo/issues`);
+        console.log(reason);
+      }
+      console.log();
+      if (!program.opts().quiet) {
+        await shouldUpdate();
+      }
+      process.exit(1);
+    }
+  });
 
 function info(message: string) {
   if (!program.opts().quiet) {
     console.log(message);
   }
 }
 
-async function runAsync(projectDirectory: string): Promise<void> {
+async function runAsync(projectDirectory?: string): Promise<void> {
```

Comment:
```suggestion
async function runAsync(maybeProjectDirectory?: string): Promise<void> {
```

i'd prefer this kind of naming for readability